tests/tests_fabric/loggers/test_csv.py

```diff
@@ -21,23 +21,20 @@
 from lightning.fabric.loggers.csv_logs import _ExperimentWriter
 
 
-def test_file_logger_automatic_versioning(tmpdir):
+def test_file_logger_automatic_versioning(tmp_path):
     """Verify that automatic versioning works."""
-    root_dir = tmpdir.mkdir("exp")
-    root_dir.mkdir("version_0")
-    root_dir.mkdir("version_1")
-    logger = CSVLogger(root_dir=root_dir, name="exp")
+    os.makedirs(tmp_path / "exp" / "version_0")
```

Comment:
Since you updated the test to use `tmp_path`, you can now use https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir